js/tree.js

```diff
@@ -372,11 +503,36 @@
 	};
 
 	$.fn.tree.defaults = {
-		dataSource: function (options, callback) {},
+		dataSource: function dataSource(options, callback) {},
 		multiSelect: false,
 		cacheItems: true,
 		folderSelect: true,
-		itemSelect: true
+		itemSelect: true,
+		/*
+		* currently calling "open" again on a folder will close it. Setting
+		* `ignoreRedundantOpens` to `true` will make the folder instead ignore
+		* the redundant call and stay open.
+		*/
+		ignoreRedundantOpens: false,
+		/*
+		* this is not necessarily indicative of how many layers deep the tree goes,
+		* only how many times `discloseAll` should be called before a warning trigger
+		* is fired for the dev to hook into. `discloseAll` will continue running
+		* after this many recursions unles the dev manually kilss it by setting
+		* `keep-disclosing` to `false`:
```

Comment:
My thought was that we would have an infinite limit by default. I guess your way _works_, just not the way I'd personally expect, so from that perspective I'll leave it up to you